### PR TITLE
Define array2field and field2array

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR moves some of the `array2field` and `field2array` methods into ClimaCore (from ClimaAtmos). I also added some simple tests.

### CODE REVIEW

File: src/Fields/Fields.jl [hunk 1]
```diff
@@ -582,4 +580,52 @@ if VERSION < v"1.10"
...
+```
+
+The dimensions of the resulting array are `([number of vertical nodes], number
+of horizontal nodes)`. Also, `field` must be a `Field` of scalars, so that the
+element type of the array is the same as the struct type of `field`.
+"""
+array2field(array, space) = Field(
+    DataLayouts.array2data(array, Spaces.local_geometry_data(space)),
+    space,
+)
```
Comment (Sbozzolo) [inline src/Fields/Fields.jl:601]: This docstring is mismatched with the actual function

File: src/DataLayouts/DataLayouts.jl [hunk 2]
```diff
@@ -1481,4 +1485,52 @@ get_Nij(::IFH{S, Nij}) where {S, Nij} = Nij
...
+
+The dimensions of the resulting array are
+ - `([number of vertical nodes], number of horizontal nodes)`.
+
+Also, this assumes that `eltype(data) <: Real`.
+"""
+function data2array end
+
+data2array(data::Union{IF, IFH}) = reshape(parent(data), :)
+data2array(data::Union{IJF, IJFH}) = reshape(parent(data), :)
```
Comment (Sbozzolo) [inline src/DataLayouts/DataLayouts.jl:1503]: Are these two functions being tested?

File: src/Fields/Fields.jl [hunk 3]
```diff
@@ -582,4 +580,52 @@ if VERSION < v"1.10"
...
+`array`. If this restriction were removed, one would also need to pass the
+desired `Field` struct type as an argument to `array2field`, which would then
+need to permute the dimensions of `array` to match the target `DataLayout`.
+"""
+function field2array(field::Field)
+    if sizeof(eltype(field)) != sizeof(eltype(parent(field)))
+        f_axis_size = sizeof(eltype(parent(field))) ÷ sizeof(eltype(field))
+        error("unable to use field2array because each Field element is \
+               represented by $f_axis_size array elements (must be 1)")
+    end
```
Comment (Sbozzolo) [inline src/Fields/Fields.jl:627]: Would it be easy to add a test for this error?

**Review by Sbozzolo — APPROVED** [review]
Thanks, I think there's a couple of functions that are not tested and two docstrings that are swapped, but everything else looks good to me.

File: src/Fields/Fields.jl (hunk 1 above)
Comment (charleskawczynski) [reply src/Fields/Fields.jl:601]: Ah, yeah, bummer, I’ll fix it

File: src/DataLayouts/DataLayouts.jl (hunk 2 above)
Comment (charleskawczynski) [reply src/DataLayouts/DataLayouts.jl:1503]: I believe so, IIRC

Comment (charleskawczynski) [reply src/DataLayouts/DataLayouts.jl:1503]: Ah, actually, no. I'll add some (there were none in ClimaAtmos)

File: src/Fields/Fields.jl (hunk 3 above)
Comment (charleskawczynski) [reply src/Fields/Fields.jl:627]: Yes, but I ran out of time this evening.

**Pull request merged**